### DISCUSSION TITLE
Column filtering support

### DISCUSSION THREAD

**Discussion by icflorescu** [icflorescu/mantine-datatable#23] [Ideas]
As discussed in Mantine's Discord channel,
> there’s simply no way for `Mantine DataTable` to accommodate every possible usage scenario out there

In my experience, in most real-life projects you’d have to deal with filtering **in parallel** with pagination, sorting, asynchronous data loading, etc., and a simple *Excel-like* column filtering implementation wouldn't bring much value.

However, it seems like a lot of people would like to see **more column filtering support baked into this library**.

So, let's discuss it here; let's try to define and implement a minimal set of useful features and see how we could expand them in time.

As I already mentioned on a number of occasions, so far I am the only developer of `Mantine DataTable`, it's one of my dearest open-source projects and I enjoy working on it, but I could also use your help with coding new valuable features.
I'm more than open to pull-requests :-)

**Comment by otobot1** [comment]
I appreciate the invite to chat here, and I totally agree with not being able to accommodate every use case. That being said, I think it would be valuable to have a multi-column search baked into the table, as I think it would be a bit of a pain as an end user to nicely integrate multi-column search into the table visually. I'm pretty new to webdev though, so I could just be not thinking of an obvious solution. For the website I'm building (for a video game community), I had envisioned having a multi-column search where there was a little button in each column header, to the left of the sort button, where users could click and then a modal would pop up with search options in it for that column. A nice second option would be just a text input beneath the column name where users could type in a search string.

**Reply by icflorescu** [reply]
I think we could reasonably accommodate this part:
> a little button in each column header, to the left of the sort button, where users could click and then a modal would pop up with search options in it for that column

We could probably show a [Mantine Popover](https://mantine.dev/core/popover/), but its content and and what actually happens then the user fills/ticks the options inside should be completely configurable, because filtering **should work in conjunction with async data loading, sorting and pagination**, so we can't just filter the already provided records array.

**Reply by otobot1** [reply]
Makes sense. Adding the popover would be more than sufficient for my use case. I'll see if I can find time to submit a PR for this in the next few days.

**Reply by otobot1** [reply]
I'm working on #40.

**Comment by pjgoncalves** [comment]
Hey there 👋 

Since this feature is not yet implemented, is there a way for me to render a text input in the table's first row so I can use it to search/filter by that attribute? 
e.g: 
![image](https://user-images.githubusercontent.com/510664/194368640-c952d766-33a0-4239-abda-9191fd565611.png)

**Reply by otobot1** [reply]
I don't believe that's possible at this time. I believe something similar was mentioned in a discussion a week or two ago, but I can't seem to find it right now. #40 is partially done, but I haven't had any spare brain power for a few weeks to get it finished. If you feel up to it, please feel free to take a look!
